### PR TITLE
Change all Moq entries to NSubstitute

### PR DESCRIPTION
**Pull request by BahdanSerabrakou**

### CODE REVIEW

File: Directory.Packages.props [hunk 1]
```diff
@@ -10,6 +10,7 @@
     <PackageVersion Include="Microsoft.CST.DevSkim" Version="1.0.22" />
     <PackageVersion Include="Microsoft.NET.Test.Sdk" Version="17.7.2" />
     <PackageVersion Include="Moq" Version="4.18.4" />
```
Comment (Eylnim) [inline Directory.Packages.props]: Please, remove all Moq references :)

File: tests/Mafin.Web.Api.Rest.Tests.Unit/MafinHttpClientBuilderTests.cs [hunk 2]
```diff
@@ -36,33 +38,33 @@ public void WithAuthHandler_WhenHandlerPassed_ShouldDelegateHttpCallToSameHandle
...
-        var client = _builder.WithAuthHandler(authHandlerMock.Object).Build();
+        var client = _builder.WithAuthHandler(authHandlerMock).Build();
 
-        _ = client.Send(requestMock.Object);
+        _ = client.Send(requestMock);
 
-        authHandlerMock.Protected().Verify(sendMethodName, Times.Once(), requestMock.Object, ItExpr.IsAny<CancellationToken>());
+        Received.InOrder(() =>
+        {
+            typeof(HttpClientHandler).InvokeMember(sendMethodName,BindingFlags.Instance | BindingFlags.NonPublic | BindingFlags.InvokeMethod, null, authHandlerMock, new object[] { requestMock, Arg.Any<CancellationToken>() });
```
Comment (Eylnim) [inline tests/Mafin.Web.Api.Rest.Tests.Unit/MafinHttpClientBuilderTests.cs]: Please, fix the warnings in this line.
Also, I see it being used twice.
Do you think it's possible to extract it, for example to a local function?

File: Directory.Packages.props (hunk 1 above)
Comment (BahdanSerabrakou) [reply Directory.Packages.props]: Fixed

File: tests/Mafin.Web.Api.Rest.Tests.Unit/MafinHttpClientBuilderTests.cs (hunk 2 above)
Comment (BahdanSerabrakou) [reply tests/Mafin.Web.Api.Rest.Tests.Unit/MafinHttpClientBuilderTests.cs]: Fixed

**Comment by Molnix888** [comment]
Note to self: check UnsafeAccessor application for mocking protected class members.

**Review by Molnix888 — APPROVED** [review]

**Pull request merged**